pkg/cluster/root_app.go

```diff
@@ -21,6 +21,7 @@ func ConstructRootApp(
 	clusterSpecCm *corev1.ConfigMap, // nil for gen2
 	profileName string,
 	managementClusterUrl string,
+	withCAS bool, // false during gen1 cluster update
```

Comment:
Since this only applies to gen2, rename it to something like `casForGen2` to avoid confusion.